docker-compose.yml

```diff
@@ -103,6 +103,32 @@ services:
     volumes:
       - ./build/certs:/etc/certs
       - ./build/mongod.conf:/etc/mongod.conf
+  
+  # used for testing a replica set
+  mongodb_replset:
+    build:
+      context: ./build/deps
+      dockerfile: ${MONGO_DOCKERFILE:-mongo6}.Dockerfile
+    container_name: mongodb_replset
+    command: mongod --port 27017 --bind_ip_all --replSet rs0
+    ports:
+      - 47017:27017
+    extra_hosts:
+    - "host.docker.internal:host-gateway"
+    environment:
+      # Always UTC+05:45. Set to catch timezone problems.
+      - TZ=Asia/Kathmandu
+  
+  # initiates the replica set
+  init_mongodb_replset:
```

Comment:
I'm lacking understanding why we are separating `mongodb_replset` and `init_mongodb_replset`. The reason can be documented to help us understand why :hugs:

Reply:
I don't think it's possible to run [`rs.initiate()`](https://www.mongodb.com/docs/manual/reference/method/rs.initiate/) from the same container.

Reply:
Added a comment in `docker-compose.yml` as well